src/cpp/src/visual_language/inputs_embedder.hpp

```diff
@@ -114,6 +120,8 @@ class InputsEmbedder {
         // True if chat mode is activated to save conversation
         // history between generate() calls.
         bool m_is_chat_conversation = false;
+        // if True, full history will be used as prompt on each chat generation
+        bool m_use_full_chat_history = false;
```

Comment:
Why does the embedder manage that flag. The implementation is fully in `VLMPipelineImpl`. Move it there

Reply:
Moved.